loki/tests/test_subroutine.py

```diff
@@ -1178,15 +1178,15 @@ def test_external_stmt(tmp_path, frontend):
     routine = source['routine_external_stmt']
     assert len(routine.arguments) == 8
 
-    for decl in FindNodes(VariableDeclaration).visit(routine.spec):
+    for decl in FindNodes(ir.VariableDeclaration).visit(routine.spec):
         # Skip local variables
         if decl.symbols[0].name in ('invar', 'outvar', 'tmp'):
             continue
         # Is the EXTERNAL attribute set?
         assert decl.external
         for v in decl.symbols:
             # Are procedure names represented as Scalar objects?
-            assert isinstance(v, ProcedureSymbol)
+            assert isinstance(v, sym.ProcedureSymbol)
             assert isinstance(v.type.dtype, ProcedureType)
             assert v.type.external is True
             assert v.type.dtype.procedure == BasicType.DEFERRED
```

Comment:
Codecov thinks these lines are never executed, so it looks to me like the test isn't quite doing what it's meant to do.
This isn't related to this PR but maybe you can have a quick look if you don't mind?

Reply:
Wow, good spot for codecov! These lines were indeed implicitly skipped, as the test was attempting to check `ProcedureDeclaration` nodes, but was iterating `VariableDeclaration`, likely as the former had been introduced much later than the test. I've pushed a fix that should rectify that.